zh-CN/hello-world.md

```diff
@@ -0,0 +1,506 @@
+# Hello, World
+
+**[你可以在这里找到本章的所有代码](https://github.com/quii/learn-go-with-tests/tree/master/hello-world)**
+
+当使用新的语言写你的第一个程序时，有一个传统就是写 Hello，world。创建一个 `hello.go` 的文件并写入这些代码。输入 `go run hello.go` 去运行它。
+
+```go
+package main
+
+import "fmt"
+
+func main() {
+    fmt.Println("Hello, world")
+}
+```
+
+## 它是如何工作的
+
+当你使用 Go 编写程序时，你将会有一个 `main` 的包，其中定义了 `main` 的函数。`func` 关键字用来定义带有名称和主体的函数。
+
+通过 `import "fmt"` 我们导入一个包含 `Println` 函数的包，我们用它来打印输出。
+
+## 如何测试
+
+怎么测试这个？把你的"领域"（domain）代码与外界分开可以避免副作用（side-effects）。`fmt.Println` 是一种副作用（打印到 stdout），我们发送的字符串是我们的领域（代码）。
+
+所以我们把这些问题分开，就更容易测试了
+
+```go
+package main
+
+import "fmt"
+
+func Hello() string {
+    return "Hello, world"
+}
+
+func main() {
+    fmt.Println(Hello())
+}
+```
+
+我们再次使用 `func` 创建了一个新函数，但是这次我们在定义中添加了另一个关键字 `string`。这意味着这个函数返回一个 `string`。
+
+现在创建一个名为 `hello_test.go` 的新文件，我们将在这里为 `Hello` 函数编写一个测试
+
+```go
+package main
+
+import "testing"
+
+func TestHello(t *testing.T) {
+    got := Hello()
+    want := "Hello, world"
+
+    if got != want {
+        t.Errorf("got '%s' want '%s'", got, want)
+    }
+}
+```
+
+在解释之前，让我们先运行一下代码。在终端运行 `go test`，它应该已经通过了！为了检验测试，可以通过尝试改变 `want` 字符串来破坏测试。
+
+请注意，你不必在多个测试框架之间进行选择，也不必破译测试 DSL 来编写测试。你需要的一切都内建在语言中，语法与你将要编写的其余代码相同。
+
+### 编写测试
+
+编写测试和写函数很类似，其中有一些规则
+
+- 它需要在一个名为 `xxx_test.go` 的文件中编写
+- 测试函数的命名必须从单词 `Test` 开始
+- 测试函数只接受一个参数 `t *testing.T`
+
+现在这些信息足以让我们明白，类型为 `*testing.T` 的变量 `t` 是你在测试框架中的 "hook"（钩子），所以你可以在想要失败时执行 `t.Fail()` 之类的操作。
+
+#### 新的知识
+
+`if`
+
+Go 的 `if` 语句非常类似于其他编程语言。
+
+**声明变量**
+
+我们使用语法 `varName := value` 声明了一些变量，它允许我们在测试中重用一些值以获得可读性。
+
+`t.Errorf`
```

Comment:
原文的 **if**、 **Declaring variables**、 **t.Errorf** 三部分应该是一个并列的关系，结构上同属于 `New things` 的次级标题（虽然 Markdown 并不认可这种形式的列表）。

这里把 **if** 和 **t.Errorf** 改成了 code 标签，也不能算错，但从排版上来讲就体现不出这个并列的关系了。

只是我的一个细微的发现， @polaris1119 和 @Donng 可以参考一下 @pityonline 的意见。

Reply:
这三段应该是 `New things` 的次级标题，作者用的是着重符号标记，可能是为了加粗显示。在 markdownlint 里的 [MD036 - Emphasis used instead of a header](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md036---emphasis-used-instead-of-a-header) 是不建议用这样表达标题的，鉴于这条规则不是特别通用，而且我也遇到过这种用法，所以在 `.mdlrc` 中忽略了这条规则。经 @flw-cn 提示，我认为保持与原文统一比较好。